oshi-core/src/main/java/oshi/hardware/platform/unix/solaris/SolarisComputerSystem.java

```diff
@@ -217,22 +193,29 @@ private static final class SmbiosStrings {
         private final String boardVersion;
         private final String boardSerialNumber;
 
-        private SmbiosStrings(String biosVendor, String biosVersion, String biosDate, //
-                String manufacturer, String model, String serialNumber, String uuid, //
-                String boardManufacturer, String boardModel, String boardVersion, String boardSerialNumber) {
-            this.biosVendor = Util.isBlank(biosVendor) ? Constants.UNKNOWN : biosVendor;
-            this.biosVersion = Util.isBlank(biosVersion) ? Constants.UNKNOWN : biosVersion;
-            this.biosDate = Util.isBlank(biosDate) ? Constants.UNKNOWN : biosDate;
-
-            this.manufacturer = Util.isBlank(manufacturer) ? Constants.UNKNOWN : manufacturer;
-            this.model = Util.isBlank(model) ? Constants.UNKNOWN : model;
-            this.serialNumber = Util.isBlank(serialNumber) ? Constants.UNKNOWN : serialNumber;
-            this.uuid = Util.isBlank(uuid) ? Constants.UNKNOWN : uuid;
-
-            this.boardManufacturer = Util.isBlank(boardManufacturer) ? Constants.UNKNOWN : boardManufacturer;
-            this.boardModel = Util.isBlank(boardModel) ? Constants.UNKNOWN : boardModel;
-            this.boardVersion = Util.isBlank(boardVersion) ? Constants.UNKNOWN : boardVersion;
-            this.boardSerialNumber = Util.isBlank(boardSerialNumber) ? Constants.UNKNOWN : boardSerialNumber;
+        private SmbiosStrings(Map<String, String> smbTypeBIOSStrings, Map<String, String> smbTypeSystemStrings,
+                Map<String, String> smbTypeBaseboardStrings) {
+            final String vendorMarker = "Vendor";
+            final String biosDateMarker = "Release Date";
+            final String biosVersionMarker = "VersionString";
```

Comment:
This isn't your error, but a bug I found testing your code. Should be a space between Version and String to match the sample output.

Other than that LGTM and ready to merge!

Reply:
I have updated it. Thank you :)